lints/lint_ext_tor_service_descriptor_hash_invalid.go

```diff
@@ -0,0 +1,318 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zcrypto/x509/pkix"
+	"github.com/zmap/zlint/util"
+)
+
+type torServiceDescHashInvalid struct{}
+
+func (l *torServiceDescHashInvalid) Initialize() error {
+	return nil
+}
+
+// CheckApplies returns true if the certificate is a subscriber certificate that
+// contains a subject name ending in `.onion`.
+func (l *torServiceDescHashInvalid) CheckApplies(c *x509.Certificate) bool {
+	if !util.IsSubscriberCert(c) {
+		return false
+	}
+	names := append(c.DNSNames, c.Subject.CommonName)
```

Comment:
We might want to move this into a util helper since it's already in multiple lints and could plausibly be reused in the future.